hibernate6-jpa/src/test/groovy/io/micronaut/configuration/hibernate/jpa/proxy/IntroduceHibernateProxySpec.groovy

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.configuration.hibernate.jpa.proxy
+
+import io.micronaut.aop.Introduced
+import io.micronaut.context.ApplicationContext
+import org.hibernate.proxy.HibernateProxy
+import spock.lang.AutoCleanup
+import spock.lang.Shared
+import spock.lang.Specification
+
+import jakarta.persistence.EntityManager
+import jakarta.persistence.EntityManagerFactory
+
+class IntroduceHibernateProxySpec extends Specification {
```

Comment:
I was getting some errors when trying to run this about bytecode set to none or something, hopefully will be fixed with micronaut-data or at least clearer error.
Or something changed in Hibernate6, saw Quarkus had some similar errors so I might need someone's help with it